airflow/serialization/serialized_objects.py

```diff
@@ -1103,22 +1106,26 @@ def detect_task_dependencies(task: Operator) -> list[DagDependency]:
                 DagDependency(
                     source=task.partial_kwargs["external_dag_id"],
                     target=task.dag_id,
+                    label=task.task_display_name or task.task_id,
                     dependency_type="sensor",
                     dependency_id=task.task_id,
                 )
             )
+
         for obj in task.outlets or []:
             if isinstance(obj, Asset):
                 deps.append(
                     DagDependency(
                         source=task.dag_id,
                         target="asset",
+                        label=obj.name,
                         dependency_type="asset",
-                        dependency_id=obj.name,
+                        dependency_id=None,
```

Comment:
Why don’t we give this a dependency id? We probably should since otherwise all asset and alias dependencies would have the same `node_id`.

Reply:
This needs some rework and will be removed. related to the resolving thing we discussed on Friday

Reply:
It now stores asset unique keys when serializing and fetches the real asset ID from db when calling `SerializedDagModel.get_dag_dependencies`